dsl2.nf

```diff
@@ -126,13 +127,27 @@ workflow {
     inputMapping = validate_wf.out.inputMapping
     inputPairing = validate_wf.out.inputPairing
 
+    if (params.bam2fastq)
+    {
+      inputBam = inputMapping
+      bam2fastq(inputBam)
+      fastqs = bam2fastq.out.fastqOutput
+			.map { idSample, targets, files_pe1, files_pe2
+			  -> tuple(groupKey(idSample, files_pe1.size()), targets, files_pe1, files_pe2)
```

Comment:
This fails with large bams that exceed the max int value.  The type of  files_pe1.size() would then be 'long' which is not able to be used by groupKey.

Reply:
Do you have a good purposal for this issue? @johnoooh